source/compiler-core/slang-tint-compiler.cpp

```diff
@@ -0,0 +1,162 @@
+#include "slang-tint-compiler.h"
+
+#include "slang-artifact-associated-impl.h"
+
+#include "../../external/slang-tint-headers/slang-tint.h"
+
+namespace Slang
+{
+
+    class TintDownstreamCompiler : public DownstreamCompilerBase
+    {
+
+    public:
+
+        // IDownstreamCompiler
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL compile(
+            const CompileOptions& options, IArtifact** outResult) SLANG_OVERRIDE;
+
+        virtual SLANG_NO_THROW bool SLANG_MCALL canConvert(
+            const ArtifactDesc& from, const ArtifactDesc& to) SLANG_OVERRIDE;
+
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL convert(
+            IArtifact* from, const ArtifactDesc& to, IArtifact** outArtifact)
+            SLANG_OVERRIDE;
+
+        virtual SLANG_NO_THROW bool SLANG_MCALL isFileBased() SLANG_OVERRIDE
+        {
+            return false;
+        }
+
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL getVersionString(
+            slang::IBlob** outVersionString) SLANG_OVERRIDE;
+
+        SlangResult compile(IArtifact *const sourceArtifact, IArtifact** outArtifact);
+
+        SlangResult init(ISlangSharedLibrary* library);
+
+    protected:
+
+        ComPtr<ISlangSharedLibrary> m_sharedLibrary;
+
+    private:
+
+        tint_CompileFunc m_compile;
+        tint_FreeBufferFunc m_freeBuffer;
+    };
+
+    SlangResult TintDownstreamCompiler::init(ISlangSharedLibrary* library)
+    {
+        tint_CompileFunc compile =
+            (tint_CompileFunc)library->findFuncByName("tint_compile");
+        if (compile == nullptr)
+        {
+            return SLANG_FAIL;
+        }
+
+        tint_FreeBufferFunc freeBuffer =
+            (tint_FreeBufferFunc)library->findFuncByName("tint_free_buffer");
+        if (m_freeBuffer == nullptr)
+        {
+            return SLANG_FAIL;
+        }
+
+        m_sharedLibrary = library;
+        m_desc = Desc(SLANG_PASS_THROUGH_TINT);
+        m_compile = compile;
+        m_freeBuffer = freeBuffer;
+        return SLANG_OK;
+    }
+
+    SlangResult TintDownstreamCompilerUtil::locateCompilers(
+        const String& path,
+        ISlangSharedLibraryLoader* loader,
+        DownstreamCompilerSet* set)
+    {
+        ComPtr<ISlangSharedLibrary> library;
+        SLANG_RETURN_ON_FAIL(
+            DownstreamCompilerUtil::loadSharedLibrary(
+                path, loader, nullptr, "slang-tint", library));
+        SLANG_ASSERT(library);
+
+        TintDownstreamCompiler *const compiler = new TintDownstreamCompiler();
+        SLANG_RETURN_ON_FAIL(compiler->init(library));
```

Comment:
This leaks. Use RefPtr<TintDownstreamCompiler>

Reply:
Done.

The class doesn't implement `RefObject`, which seems required for `RefPtr`. I switched to using `ComPtr<IDownstreamCompiler>` like the analogous code in slang-metal-compiler.cpp.